.github/workflows/deploy-hubs.yaml

```diff
@@ -24,25 +20,23 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Pull out any hubs that need deploying from the labels on the merge commit to staging
+      - name: Check for hubs that need deploying from the labels on the merge commit to staging
         run: |
           echo "PR labels: ${{ steps.pr-labels.outputs.labels }}"
-          HUBS=()
-          # If the PR labels "hub-images" or "jupyterhub-deployment" are present, this
-          # means the base hub image has changed, and all hubs (staging or prod) need to
-          # be redeployed.  The rest of this job will not run in that case.
-          if [ -c "${GITHUB_PR_LABEL_HUB_IMAGES}" ] || [ -c "${GITHUB_PR_LABEL_JUPYTERHUB_DEPLOYMENT}" ]; then
-            echo "Base hub image has changed, not deploying individual hubs to staging"
+          # If the PR labels "hub-images" or "jupyterhub-deployment" are
+          # present, this means the base hub image has changed, and all hubs
+          # (staging or prod) need to be redeployed.
+          #
+          # Otherwise, just check to see if the PR labels contain any hubs,
+          # and if so, deploy just those hubs to staging.
+          if [[ -n ${GITHUB_PR_LABEL_HUB_IMAGES} || -n ${GITHUB_PR_LABEL_JUPYTERHUB_DEPLOYMENT} ]]; then
```

Comment:
@felder :  sanity check here.  the github action that grabs labels from a push exports them to env vars in the format of `GITHUB_PR_LABEL_<whatever>=1`.  in this case, i want to check for the labels of `jupyterhub-deployment` and/or `hub-images` (stored as the vars in the code snippet above).

if i want to check if these are indeed set, the way i do it here (`if [[ -n ${GITHUB_PR_LABEL_HUB_IMAGES} || -n ${GITHUB_PR_LABEL_JUPYTERHUB_DEPLOYMENT} ]]`) is correct, yes?

(i did a bunch of testing on the CLI and this seems to give us what we want but a 2nd pair of eyes etc)

Reply:
@shaneknapp -n evaluates to true if the variable is not set to the empty string.  You may want to use -v instead which evaluates to true if the variable is set.  -v also expects the name of the variable as opposed to an expansion of the variable.

So you may want to try`if [[ -v GITHUB_PR_LABEL_HUB_IMAGES || -v GITHUB_PR_LABEL_JUPYTERHUB_DEPLOYMENT ]]`

Example:
```
ras-29-57:~ felder$ cat test.sh
#!/usr/bin/env bash

TEST=""

if [[ -v TEST ]]; then
  echo "TEST is set"
fi

unset TEST

if [[ ! -v TEST ]]; then
  echo "TEST is no longer set"
fi
ras-29-57:~ felder$ ./test.sh
TEST is set
TEST is no longer set
ras-29-57:~ felder$

```

Reply:
alright, lemme try that...  i'll set some labels to test this when i merge to staging